Add Algolia

kagankan opened the pull request:
Fixes #573

## What is the purpose of this PR?

- [ ] Fix bug
- [ ] Fix typo
- [ ] Update specs
- [ ] Add new rule
- [ ] Add new parser
- [ ] Improve or refactor rules
- [ ] Add a new core feature
- [ ] Improve or refactor core features
- [ ] Update documents
- [x] Others

### Description

Added Algolia search to the website.

However, we use the index of `markuplint.dev`, not `next.markuplint.dev`.
We need to add an index for `next.markuplint.dev`, and use it.

## Checklist

Fill out the checks for the applicable purpose.

- [ ] Fix bug
  - [ ] Add the test that can check whether resolved the issue.
- [ ] Update specs
  - [ ] Element
    - [ ] Update **[global attributes](https://github.com/markuplint/markuplint/blob/main/packages/%40markuplint/html-spec/src/spec-common.attributes.json)**
    - [ ] Update **[content models](https://github.com/markuplint/markuplint/blob/main/packages/%40markuplint/html-spec/src/spec-common.contents.json)**
    - [ ] Add also ARIA that you explored **ARIA in HTML [Recommendation](https://www.w3.org/TR/html-aria/) and [Editor's draft](https://w3c.github.io/html-aria/)**
  - [ ] Attribute
    - [ ] Update **[global attributes](https://github.com/markuplint/markuplint/blob/main/packages/%40markuplint/html-spec/src/spec-common.attributes.json)**
    - [ ] Update **[IDL attribute](https://github.com/markuplint/markuplint/blob/main/packages/%40markuplint/parser-utils/src/idl-attributes.ts)**
- [ ] Add new rule
  - [ ] Add a test
  - [ ] Add a README document
    - [ ] Create an issue that requests translating the doc if you want.
  - [ ] Add to [index](https://github.com/markuplint/markuplint/blob/main/packages/%40markuplint/rules/src/index.ts)
  - [ ] Add JSON schema to [schema.json](https://github.com/markuplint/markuplint/blob/main/packages/%40markuplint/rules/schema.json)
  - [ ] Add to [initializer](https://github.com/markuplint/markuplint/blob/main/packages/markuplint/src/cli/init/index.ts)
  - [ ] Add to [`@markuplint/config-presets`](https://github.com/markuplint/markuplint/tree/main/packages/%40markuplint/config-presets)
- [ ] Add new parser
  - [ ] Add a test
    - [ ] Do a test with some parser.
  - [ ] Add to [initializer](https://github.com/markuplint/markuplint/blob/main/packages/markuplint/src/cli/init/index.ts)

YusukeHirao commented:
@kagankan 

> I have added the domain "next.markuplint.dev" to the application SIO9QLVJBB. 
>
> Now you should be able to crawl this domain, so please update your crawler configuration.
>
> You might want to update parameters like startUrls, sitemaps, and pathsToMatch in the configuration. You can find a generic configuration example on [this page](https://docsearch.algolia.com/docs/required-configuration#the-generic-configuration-example).

Added `next.markuplint.dev` to the app.
Please continue.

Thanks.

kagankan commented:
@YusukeHirao 
Updated this PR.
We can search docs in the website!

## Examples and Features

The words are searched in the page title, the headings, and the content.
![image](https://user-images.githubusercontent.com/34136752/205432637-75dd4e30-24d0-47d3-b1f2-a24c8667560c.png)

Algolia tolerates typos
https://www.algolia.com/doc/guides/managing-results/optimize-search-results/typo-tolerance/
![image](https://user-images.githubusercontent.com/34136752/205432492-cdca4241-e4c7-4d8a-8aab-704df9356ff3.png)

The `td` in first column is set to be treated as a heading.
(This is the default setting.)
![image](https://user-images.githubusercontent.com/34136752/205432857-d772e96f-dc4c-46b0-9c77-1000d867355d.png)
![image](https://user-images.githubusercontent.com/34136752/205432870-1c347104-e07a-4a65-827a-b500137f8235.png)

These Algolia settings are configurable.

## Other changes

- changed URL for next.markuplint.dev
- changed editUrl for dev branch
- summarized settings that should be changed when deployment settings change

Please check it.

YusukeHirao commented:
@kagankan 

Awosome!!
Thank you!!
